src/openldap/mod.rs

```diff
@@ -0,0 +1,531 @@
+            username: username.to_string(),
+            password: password.to_string(),
+        });
+        self
+    }
+
+    /// Add users (can be called multiple times)
+    /// Default: `[]`
+    ///
+    /// Alternatively, users can be added via [`OpenLDAP::with_user`].
+    pub fn with_users<Username: ToString, Password: ToString>(
+        mut self,
+        user_password: impl IntoIterator<Item = (Username, Password)>,
+    ) -> Self {
+        for (username, password) in user_password.into_iter() {
+            self.users.push(User {
+                username: username.to_string(),
+                password: password.to_string(),
+            })
+        }
+        self
+    }
+
+    /// Sets the users' dc
+    /// Default: `"users"`
+    pub fn with_users_dc(mut self, user_dc: impl ToString) -> Self {
+        self.env_vars
+            .insert("LDAP_USER_DC".to_owned(), user_dc.to_string());
+        self
+    }
+
+    /// Sets the users' group
+    /// Default: `"readers"`
+    pub fn with_users_group(mut self, users_group: impl ToString) -> Self {
+        self.env_vars
+            .insert("LDAP_GROUP".to_owned(), users_group.to_string());
+        self
+    }
+
+    /// Extra schemas to add, among [`OpenLDAP`]'s distributed schemas.
+    /// Default: `["cosine", "inetorgperson", "nis"]`
+    pub fn with_extra_schemas<S: ToString>(
+        mut self,
+        extra_schemas: impl IntoIterator<Item = S>,
+    ) -> Self {
+        self.env_vars
+            .insert("LDAP_ADD_SCHEMAS".to_owned(), "yes".to_owned());
+        let extra_schemas = extra_schemas
+            .into_iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>()
+            .join(", ");
+        self.env_vars
+            .insert("LDAP_EXTRA_SCHEMAS".to_owned(), extra_schemas);
+        self
+    }
+
+    /// Allow anonymous bindings to the LDAP server.
+    /// Default: `true`
+    pub fn with_allow_anon_binding(mut self, allow_anon_binding: bool) -> Self {
+        if allow_anon_binding {
+            self.env_vars
+                .insert("LDAP_ALLOW_ANON_BINDING".to_owned(), "yes".to_owned());
+        } else {
+            self.env_vars
+                .insert("LDAP_ALLOW_ANON_BINDING".to_owned(), "no".to_owned());
+        }
+        self
+    }
+
+    /// Set hash to be used in generation of user passwords.
+    /// Default: [`PasswordHash::SSHA`].
+    pub fn with_ldap_password_hash(mut self, password_hash: PasswordHash) -> Self {
+        self.env_vars.insert(
+            "LDAP_PASSWORD_HASH".to_owned(),
+            match password_hash {
+                PasswordHash::SSHA => "{SSHA}".to_owned(),
+                PasswordHash::SHA => "{SHA}".to_owned(),
+                PasswordHash::SMD5 => "{SMD5}".to_owned(),
+                PasswordHash::MD5 => "{MD5}".to_owned(),
+                PasswordHash::CRYPT => "{CRYPT}".to_owned(),
+                PasswordHash::CLEARTEXT => "{CLEARTEXT}".to_owned(),
```

Comment:
Minor, but we also can utilize parse display here:

```rs
#[derive(parse_display::Display)]
#[display("{{{}}}")] // it's escaped curly braces
enum PasswordHash { ... }
```

Reply:
The reason why I did not include this is that I thought that this is too obvuscated.
But fair, is included ^^